### ISSUE TITLE
Implement support for library tests using CG2-compiled framework and add the corresponding CI run

### ISSUE THREAD

**Issue by trylek** [dotnet/runtime#50127]
As part of the switch-over from Crossgen1 to Crossgen2 we should start running regular library test runs over CG2-compiled framework to catch potential regressions or issues. We should update the library test build scripts to optionally support CG2 compilation of framework assemblies, add the corresponding support to the YAML scripts and define a new type of run leveraging this functionality to run library tests over CG2-compiled framework. For now I would run them on a daily basis; if they turn out to be stable enough, we may consider switching some of the library tests in PR runs to work this this too.

/cc @dotnet/runtime-infrastructure, @dotnet/crossgen-contrib

**Comment by Anipik** [comment]
@trylek can we move this to 7.0.0 ?

**Comment by davidwrighton** [comment]
Yes. We're not going to do this in the 6.0 timeframe.

**Comment by trylek** [comment]
I don't think this is strictly necessary in 7.0. It's kind of nice to have but, as it's an infrastructure change, it doesn't need to be tightly coupled with a particular release. In fact I believe this to be the perfect candidate for a quality week project.
